test/qpy_compat/test_qpy.py

```diff
@@ -288,6 +288,58 @@ def generate_evolution_gate():
     return qc
 
 
+def generate_control_flow_circuits():
+    """Test qpy serialization with control flow instructions."""
+    from qiskit.circuit.controlflow import WhileLoopOp
+
+    # If instruction
+    circuits = []
+    qc = QuantumCircuit(2, 2)
+    qc.h(0)
+    qc.measure(0, 0)
+    with qc.if_test((qc.clbits[0], True)):
+        qc.x(1)
+    qc.measure(1, 1)
+    circuits.append(qc)
+    # If else instruction
+    qc = QuantumCircuit(2, 2)
+    qc.h(0)
+    qc.measure(0, 0)
+    with qc.if_test((qc.clbits[0], True)) as else_:
+        qc.x(1)
+    with else_:
+        qc.y(1)
+    qc.measure(1, 1)
+    circuits.append(qc)
+    # While loop
+    qc = QuantumCircuit(2, 1)
+    block = QuantumCircuit(2, 1)
+    block.h(0)
+    block.cx(0, 1)
+    block.measure(0, 0)
+    while_loop = WhileLoopOp((qc.clbits[0], 0), block)
+    qc.append(while_loop, [0, 1], [0])
+    circuits.append(qc)
+    # for loop range
+    qc = QuantumCircuit(2, 1)
+
+    with qc.for_loop(range(5)):
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.measure(0, 0)
+        qc.break_loop().c_if(0, True)
```

Comment:
builder

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7584/commits/261663865ff09c6243f9f3d6bbd22550189d7648